lib/http/request.rb

```diff
@@ -106,7 +106,7 @@ def redirect(uri, verb = @verb)
 
     # Stream the request to a socket
     def stream(socket)
-      include_proxy_headers if using_proxy? && !@uri.https?
```

Comment:
Oh, I think you're correct. https://github.com/httprb/http/blob/master/lib/http/request.rb#L161